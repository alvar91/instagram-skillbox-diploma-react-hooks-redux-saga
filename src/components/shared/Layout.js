import React from 'react';
import PropTypes from 'prop-types';
import { useLayoutStyles } from '../../styles';
import SEO from '../shared/Seo';
import Navbar from '../shared/Navbar';

function Layout({ children, minimalNavbar = false, title, marginTop = 60 }) {
  const classes = useLayoutStyles();

  return (
    <section className={classes.section}>
      <SEO title={title} />
      <Navbar minimalNavbar={minimalNavbar} />
      <main className={classes.main} style={{ marginTop }}>
        <section className={classes.childrenWrapper}>
          <div className={classes.children}>{children}</div>
        </section>
      </main>
    </section>
  );
}

Layout.propTypes = {
  children: PropTypes.any.isRequired,
  minimalNavbar: PropTypes.bool,
  title: PropTypes.string,
  marginTop: PropTypes.number,
};

export default React.memo(Layout);
