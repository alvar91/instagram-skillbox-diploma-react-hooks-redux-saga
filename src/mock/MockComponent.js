import React from "react";

function MockComponent() {
  return <h2>Mock Component</h2>;
}

export default MockComponent;
