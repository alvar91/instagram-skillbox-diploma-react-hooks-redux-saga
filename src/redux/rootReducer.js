// Core
import { combineReducers } from 'redux';

// Instruments
import { postsReducer } from './posts/reducer';
import { authReducer } from "./auth/reducer";

export const rootReducer = combineReducers({
    posts: postsReducer,
    auth: authReducer,
});
