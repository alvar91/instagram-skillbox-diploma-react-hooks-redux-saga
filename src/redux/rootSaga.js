// Core
import { all, call } from "redux-saga/effects";

// Watchers
import { watchPosts } from "./posts/saga/watchers";
import { watchAuth } from "./auth/saga/watchers";

export function* rootSaga() {
  yield all([call(watchPosts), call(watchAuth)]);
}
