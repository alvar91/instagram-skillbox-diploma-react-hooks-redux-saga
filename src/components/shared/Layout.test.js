// Core
import React from "react";
import renderer from "react-test-renderer";
import Layout from "./Layout";
import MockComponent from "../../mock/MockComponent";
import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";
import { store } from "../../redux/store";

import { configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

/**
 * Setup function for Layout component.
 * @param {object} props
 * @returns {RenderTree}
 */
const setupRenderTree = (props) => {
  return renderer.create(
    <Provider store={store}>
      <MemoryRouter initialEntries={[`/`]}>
        <Layout {...props}>
          <MockComponent />
        </Layout>
      </MemoryRouter>
    </Provider>
  );
};

/**
 * Setup function for Layout component.
 * @param {object} props
 * @returns {ReactRenderer}
 */
const setupTestRenderer = (props) => {
  return mount(
    <Provider store={store}>
      <MemoryRouter initialEntries={[`/`]}>
        <Layout {...props}>
          <MockComponent />
        </Layout>
      </MemoryRouter>
    </Provider>
  );
};

describe("Layout", () => {
  let tree;
  let testRenderer;
  let component;
  const props = { title: "Test title" };
  beforeAll(() => {
    tree = setupRenderTree(props);
    testRenderer = setupTestRenderer(props);
    component = testRenderer.find(Layout);
  });
  it("Layout should correctly renders markup", () => {
    expect(tree).toMatchSnapshot();
  });
  it("Layout should contain correct props", () => {
    expect(component).toHaveLength(1);
    expect(component.props()).toMatchObject(props);
  });
});
