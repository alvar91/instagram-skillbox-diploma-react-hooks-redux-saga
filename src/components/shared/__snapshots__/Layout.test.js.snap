// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Layout should correctly renders markup 1`] = `
<section
  className="makeStyles-section-1"
>
  <div
    className="makeStyles-progressContainer-29"
    style={
      Object {
        "opacity": 1,
        "transition": "opacity 200ms linear",
      }
    }
  >
    <div
      className="makeStyles-progressBar-28"
      style={
        Object {
          "marginLeft": "0%",
          "transition": "margin-left 200ms linear",
        }
      }
    >
      <div
        className="makeStyles-progressBackground-30"
      />
    </div>
  </div>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-5 mui-fixed MuiPaper-elevation4"
  >
    <section
      className="makeStyles-section-6"
    >
      <div
        className="makeStyles-logoContainer-7"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <div
            className="makeStyles-logoWrapper-8"
          >
            <img
              alt="Instagram"
              className="makeStyles-logo-9"
              src="logo.png"
            />
          </div>
        </a>
      </div>
      <div
        className="makeStyles-linksContainer-11"
      >
        <div
          className="makeStyles-linksWrapper-12"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            <svg
              aria-label="Home"
              className="_8-yf5"
              fill="#262626"
              height={22}
              viewBox="0 0 48 48"
              width={22}
            >
              <path
                d="M45.5 48H30.1c-.8 0-1.5-.7-1.5-1.5V34.2c0-2.6-2.1-4.6-4.6-4.6s-4.6 2.1-4.6 4.6v12.3c0 .8-.7 1.5-1.5 1.5H2.5c-.8 0-1.5-.7-1.5-1.5V23c0-.4.2-.8.4-1.1L22.9.4c.6-.6 1.6-.6 2.1 0l21.5 21.5c.3.3.4.7.4 1.1v23.5c.1.8-.6 1.5-1.4 1.5z"
              />
            </svg>
          </a>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Login
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </section>
  </header>
  <main
    className="makeStyles-main-2"
    style={
      Object {
        "marginTop": 60,
      }
    }
  >
    <section
      className="makeStyles-childrenWrapper-3"
    >
      <div
        className="makeStyles-children-4"
      >
        <h2>
          Mock Component
        </h2>
      </div>
    </section>
  </main>
</section>
`;
